google/cloud/forseti/services/inventory/base/cai_gcp_client.py

```diff
@@ -109,47 +109,66 @@ def session(self):
         self._local.cai_session = db.create_readonly_session(engine=self.engine)
         return self._local.cai_session
 
-    def fetch_bigquery_iam_policy(self, project_number, dataset_id):
+    def fetch_bigquery_iam_policy(self, project_id, project_number, dataset_id):
         """Gets IAM policy of a bigquery dataset from Cloud Asset data.
 
         Args:
+            project_id (str): id of the project to query.
             project_number (str): number of the project to query.
             dataset_id (str): id of the dataset to query.
 
         Returns:
             dict: Dataset IAM Policy.
         """
+        bigquery_prefix = '//bigquery.googleapis.com/projects/{}/datasets/{}'
+
+        if project_id:
```

Comment:
At this point the code won't know if the data in the database uses one format or the other. What is needed is for the fetch_cai_asset() call to use the project_id format, and if that returns no results, the call it again using the project_number format. If there are still no results, then return {}.

Same for fetch_bigquery_dataset_policy.

Reply:
Updated.